ec/ecresource/deploymentresource/delete.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 // Delete shuts down and deletes the remote deployment.
-func delete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func deleteFunc(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Why is this being changed? `delete` was fine?

Reply:
there is a function in expanders.go:172-176 which uses the built-in `delete` function to delete keys of a map:

```go
func unsetTopology(rawRes []interface{}, d *schema.ResourceData) {
	for _, r := range rawRes {
		delete(r.(map[string]interface{}), "topology")
	}
}
```

Since we need to use this and the private `delete` function was shadowing the built-in `delete` I renamed it here.

Reply:
Ah! of course! How about changing them all to `deleteResource`, `createResource`... (or `createDeployment`) etc? It makes me cringe a little to have `read`, `create`, `update`, `deleteFunc` 😄

Reply:
haha okay the `<action>Resource` or `<action>Func` sound fine to me.